sdks/csharp/src/WebSocket.cs

```diff
@@ -167,19 +167,24 @@ public async Task Connect(string? auth, string host, string nameOrAddress, Conne
             {
                 uri += "&light=true";
             }
-            var url = new Uri(uri);
             Ws.Options.AddSubProtocol(_options.Protocol);
 
             var source = new CancellationTokenSource(10000);
             if (!string.IsNullOrEmpty(auth))
             {
-                Ws.Options.SetRequestHeader("Authorization", $"Bearer {auth}");
-            }
-            else
-            {
-                Ws.Options.UseDefaultCredentials = true;
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("Browser")))
+                {
+                    // For WebAssembly, we need to pass the auth token as a query parameter
+                    uri += $"&token={Uri.EscapeDataString(auth)}";
```

Comment:
It's generally not considered best practice to send an auth token via a URL param. To get around this in our C# (WebGL) and Typescript SDKs we generate a short-lived token which you can generate via an API call. This short lived token is then used instead of your real token. You can see an implementation of this here: (C# SDK) https://github.com/clockworklabs/SpacetimeDB/pull/2988/files and then also here for the typescript SDK: https://github.com/clockworklabs/SpacetimeDB/blob/36f7ca2583a7491285460454839e41ff43ec27a7/sdks/typescript/packages/sdk/src/websocket_decompress_adapter.ts#L93C1-L110C6

I think likely wait for #2988 to land first and see if we can just use the same auth flow that we're using for WebGL?